### ISSUE TITLE
bt-mqtt-gateway: configuration not found

### ISSUE THREAD

**Issue by Micha70** [FaserF/hassio-addons#86]
**Describe the bug**
after installation of bt-mqtt-gateway configuration not found.
Protocoll output: 
```
There is no /share/bt-mqtt-gateway.yaml file! Please edit /share/bt-mqtt-gateway.yaml.example and rename it afterwards.
Do edit it before restarting!!!!!!!

```
**To Reproduce**
1. Install bt-mqtt-gateway on home assistant
2. place configuration in /usr/share/bt-mqtt-gateway.yaml
3. start addon
4. addon does not start. Output "There is no /share/bt-mqtt-gateway.yaml file! Please edit /share/bt-mqtt-gateway.yaml.example and rename it afterwards.
Do edit it before restarting!!!!!!!"

Did already several tries with different locations for bt-mqtt-gateway.yaml:
- /usr/share/bt-mqtt-gateway.yaml
- /usr/share/hassio/homeassistant
- /usr/share/hassio/homeassistant/www
- /usr/share/hassio/homeassistant/share
- /usr/share/hassio/share

**Expected behavior**
addon starts without error message

**What Addon?**
- Addon Name: bt-mqtt-gateway
- Addon Version: 1.0.4

**Comment by FaserF** [comment]
I am unable to reproduce the issue. I am using my custom config at /share/bt-mqtt-gateway.yaml

Please note that this addon can only access files within the HA /share/ folder.
I think you are running a docker/supervised installation? Then your /share folder should be a folder up from where /config is

For example with my HA OS installation there is root (/) and it has the following folders for example:
/config
/share
/addons
/media

etc.
So please double check that you are using the correct HA /share/ folder
As I am not running supervised I am not sure, but it could be "/usr/share/hassio/homeassistant/share" if there is also a "/usr/share/hassio/homeassistant/config"

Then choose the full file directory:
/usr/share/hassio/homeassistant/share/bt-mqtt-gateway.yaml

You can also search for the dummy file created: bt-mqtt-gateway.yaml.example
The new yaml config file needs to be in the same directory

**Comment by Micha70** [comment]
Ok, see now I should address to zewelor :smile:

**Comment by FaserF** [comment]
You can check the permissions of the file before submitting to him. I will try it tomorrow with a similar config at my setup

**Comment by jhass** [comment]
I'm getting the same is a directory error. One thing I noticed is that while your start script copies the example configuration file from `/application/config.yaml.example`, on the way back it copies the configuration file to `/config.yaml`, while the error complains about `/application/config.yaml`. So maybe not an upstream issue after all?

**Comment by jhass** [comment]
This seems like a regression from https://github.com/zewelor/bt-mqtt-gateway/commit/9e6eb271d2eb031695144f6bc864ff90c8edccff#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R19 more or less.

**Comment by jhass** [comment]
I hacked around this by adding `sed -i -e 's|"config.yaml"|"../config.yaml"|' ./config.py`  to `start.sh` for now. Proper fixes I can imagine:

* Get upstream to remove the `VOLUME` directive again. It seems pretty superfluous and not required for the related addition of the bind mount to that location in their docker compose config
* Get upstream to make the ` config.yaml`  location a command-line parameter, then always pass it in explicitly here (this could avoid having to copy around stuff at all)
* Don't depend on upstream's docker image and include an independent `Dockerfile` here.

**Comment by zewelor** [comment]
I've removed VOLUME from bt-mqtt-gateway Dockerfile. https://github.com/zewelor/bt-mqtt-gateway/commit/4912e1685b31fc2543babb79f96ef494eaaa8f87

**Comment by wjarka** [comment]
> I'm getting the same is a directory error. One thing I noticed is that while your start script copies the example configuration file from `/application/config.yaml.example`, on the way back it copies the configuration file to `/config.yaml`, while the error complains about `/application/config.yaml`. So maybe not an upstream issue after all?

I am still experiencing this. Should that already work out of the box or any action required on my end? Or @FaserF's?

**Comment by FaserF** [comment]
There will be most likely action needed from my side for the docker file. I will try looking into it next week.

**Comment by wjarka** [comment]
> There will be most likely action needed from my side for the docker file. I will try looking into it next week.

Thanks man!

**Comment by FaserF** [comment]
issue has been fixed with Version 1.1.0
If there are still issues about the config file left please let me know.

https://github.com/zewelor/bt-mqtt-gateway/issues/260 can also be closed now. Thanks to @zewelor for his quick trial to fix it on his end.
